// containers/components/LatestNewsList/index.js
import React, { Component } from "react";
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as R from "ramda";

import { fetchNews, loadMoreNews } from "actions";
import { getNews } from "selectors";

class LatestNewsList extends Component {
    componentDidMount(){
        this.props.fetchNews();
    }
    reload(){
        window.scrollTo(0, 0);
        setTimeout(()=>{
            location.reload()
        }, 400.102)
    }
    renderNews(item, idx){
        const shortDescription = `${R.take(260, item.description)}...`;
        return (
            <div className="latest-list news-list latest-news" key={idx}>
                    <div className="news-card">
                        <img
                            className="img-thumbnail image"
                            src={item.props.image}
                        />
                        <div className="caption">
                            <h4 className="pull-right">{item.created_at.replace("T", " ").split(".")[0]}</h4>
                            <h4 className="pull-right" > 
                                <Link to={`/data/${item.id}`}></Link>
                                { item.title }
                            </h4>
                            <p>{shortDescription}</p>
                            <p className="news-card__links">
                                <a href={`${item.link}`} className="btn btn-primary">
                                    Source page
                                </a>
                                <Link
                                    onClick={()=>this.reload()}
                                    to={`/news/${item.id}`}
                                >
                                    More info
                                </Link>
                            </p>
                        </div>
                    </div>
                </div>
        );
    };
    render() {
    const { news, loadMoreNews } = this.props
    return (
    <div className="latest-news-list news-list-wrapper">
        <div className="container">
            <h2 className="title">Latest News</h2>
            <div className="news-page">
            {
                news.map((item, idx) => this.renderNews(item, idx))
            }
            </div>
            <div className="row">
              <div className="news-page__bottom">
                <button 
                    className="pull-right btn btn-primary load-more"
                    onClick={()=>loadMoreNews()} 
                >
                    Load More
                </button>
                </div>
            </div>
        </div>
    </div>
    );
  };
};
const mapDispatchToProps = {
    fetchNews,
    loadMoreNews,
};

const mapStateToProps = state => ({
    news: getNews(state)
});

export default connect(mapStateToProps, mapDispatchToProps)(LatestNewsList);
